.github/workflows/test-ios-device.yaml

```diff
@@ -30,27 +66,109 @@ jobs:
       - name: Set up Flutter
         uses: subosito/flutter-action@v2
         with:
-          channel: ${{ matrix.flutter-channel }}
+          flutter-version: ${{ matrix.flutter-version }}
+          cache: true
 
       - name: Preload Flutter artifacts
         run: flutter precache
 
-      - name: Run example_test.dart
-        run: ./run_ios_testlab integration_test/example_test.dart
+      - name: Set up Patrol CLI
+        working-directory: packages/patrol_cli
+        run: dart pub global activate --source path . && patrol
+
+      - name: Build tests
+        run: |
+          patrol build ios --release \
+            --exclude integration_test/notifications_test.dart \
+            --exclude integration_test/android_app_test.dart \
+            --exclude integration_test/open_app_test.dart \
+            --exclude integration_test/service_dark_mode_test.dart \
+            --exclude integration_test/service_wifi_test.dart \
+            --exclude integration_test/permissions_location_test.dart \
+            --exclude integration_test/swipe_test.dart \
+            --exclude integration_test/webview_login_test.dart \
+            --exclude integration_test/webview_leancode_test.dart
+
+      - name: Compress test files to zip
+        run: |
+          cd build/ios_integ/Build/Products
+          zip -r ios_tests.zip \
+            Release-iphoneos/Runner.app \
+            Release-iphoneos/RunnerUITests-Runner.app \
+            Runner_iphoneos16.2-arm64.xctestrun
```

Comment:
I'm worried it'll start failing once the Xcode version is updated.

Why don't you use a glob?

```bash\Runner_iphoneos*-arm64.xctestrun
```